### DISCUSSION TITLE
[API] Recovering an existing url

### DISCUSSION THREAD

**Discussion by gab12** [YOURLS/YOURLS#3764] [Q&A] [answered]
Hello,

When I create a new url via the API, but the keyword already exists, it returns an error telling me that it already exists but it doesn't give me the url.
I'd like it to return the url when a keyword already exists. How would this be possible?
Or else get a url from the keyword?

Because at the moment, when the word already exists, I haven't found any other solution other than going manually to the interface to find the url of the link.

thanks in advance

**Comment by dgw** [comment, marked as answer]
The way the `shorturl` action checks for a keyword already in use also checks if the keyword is reserved or banned, so there might not _be_ a long URL to return in this case (`error:keyword`).

If you get this error when calling the `shorturl` action, the simplest next step if you want any long URL associated with the keyword is to call the `expand` action. If the keyword is in use, the link already associated with that keyword will be returned in the `longurl` field. You will get a 404 status code if the keyword is not in use (i.e. reserved). See the API tab of `https://your.yourls.site/readme.html` for more information about available API methods.

**Comment by gab12** [comment]
I didn't thank you.
It worked so thank you for your answer, it helped me a lot and I succeeded.
I wouldn't have succeeded without you.

Especially since the documentation is not very clear and the API is not at all consistent.

The response from the shorturl action returns:
```
{
"keyword": "ozh",
"shorturl": "http:\/\/sho.rt\/1f",
"status": "success",
...
}
```

And in the expand API, when it asks for the "shorturl" parameter, you have to put the keyword and not the real "shorturl".
```
curl_setopt($ch, CURLOPT_POSTFIELDS, array( // Data to POST
'shorturl' => 'ozh',
'format' => 'json',
'action' => 'expand',
... ));

```
The API's response is also not very consistent, as the status is no longer returned in the "status" field but "message" (unlike the shorturl call)
`<message>success</message>
`
Well, the most important thing is that it works.
Thanks again a lot.

**Reply by dgw** [reply]
I'm glad I could help. 😸

Changing the API of YOURLS 1.9.x could be problematic at this point, with how many clients out there already depend on its quirks. But I'm positive the API will be more consistent in YOURLS 2.0, whenever it's ready. It's practically a from-scratch rewrite.
